src/wasm-type.h

```diff
@@ -308,6 +308,9 @@ class HeapType {
     data,
   };
   static constexpr BasicHeapType _last_basic_type = data;
+  static constexpr uintptr_t _invalid_type = -1;
```

Comment:
why not 0? (is `-1` guaranteed to segfault on use like 0 is?)

Reply:
Actually, is this needed? It would be better to avoid allowing an invalid HeapType at all. It seems like `getBasicHeapType` returns a boolean already, so we know if a value was found or not. Given that, the HeapType could be initialized to a valid value before the call? (There is a risk to doing that as well, but with a good comment there I think it's less risky than a general change to allow invalid HeapTypes to be declared as easily as `HeapType bad;`).

Reply:
Sure, I can have the default constructor just choose an arbitrary BasicHeapType. FWIW, address -1 should segfault on all reasonable platforms (as long as you're not in kernel mode or in WebAssembly or something 😉).